ambry-cloud/src/main/java/com/github/ambry/cloud/VcrServer.java

```diff
@@ -131,14 +133,21 @@ public void startup() throws InstantiationException {
       // verify the configs
       properties.verify();
 
-      virtualReplicatorCluster =
-          ((VirtualReplicatorClusterFactory) Utils.getObj(cloudConfig.virtualReplicatorClusterFactoryClass, cloudConfig,
-              clusterMapConfig, clusterMap)).getVirtualReplicatorCluster();
-
       // initialize cloud destination
       if (cloudDestinationFactory == null) {
-        cloudDestinationFactory = Utils.getObj(cloudConfig.cloudDestinationFactoryClass, properties, registry);
+        cloudDestinationFactory = Utils.getObj(cloudConfig.cloudDestinationFactoryClass, properties, registry, clusterMap);
```

Comment:
Can you point me to the correct `CloudDestinationFactory` class whose constructor takes in `clusterMap`?  I didn't find one. Is it `AzureCloudDestinationFactory`?